### PR TITLE
refactored organisations#show to reduce usage of instance variables

### PR DESCRIPTION
**Pull request by rehan-work**
https://www.pivotaltracker.com/story/show/138097221

### CODE REVIEW

**Comment by tansaku** [comment]
great start @rehan-work ! :-)

**Comment by tansaku** [comment]
hi @rehan-work - how's it going?  I was thinking this had issues, but that coveralls thing is trivial (gotta move away to a more reliable coverage system I think)

so this is ready to be merged?

**Comment by tansaku** [comment]
hi @rehan-work - any updates on this PR?

**Comment by rehan-work** [comment]
Hey Sam, I'm pretty sure this is ready to be merged, the coveralls error seems to be an issue with coveralls's code measures, seems impossible to solve

**Comment by tansaku** [comment]
thanks @rehan-work - yeah - coveralls is splitting hairs on us - great work here.

We're leaning away from controller and view specs, but you're refactoring them here and there's still significant logic they need to cover.

I think the only change I'd be tempted to make is to replace those `should`s with `is_expected_to` - but happy to make that a refactoring ticket if you're busy with other things

File: spec/controllers/organisations_controller_spec.rb [hunk 1]
```diff
@@ -494,6 +514,23 @@ def double_organisation(stubs={})
...
+
+    let!(:org) { FactoryGirl.create(:organisation)}
+
+    subject { 
+      user = create :user
+      allow(controller).to receive(:current_user).and_return(user)
+      user_opts = controller.send(:get_user_options, org)
+    }
+
+    it { should be_a_kind_of Hash }
```
Comment (tansaku) [inline spec/controllers/organisations_controller_spec.rb]: for rspec 3 I think we're supposed to prefer

```
it { is_expected_to be_a_kind_of Hash }
```
etc.?

**Comment by rehan-work** [comment]
I'll try to get this fixed up this week and check into the kickoff on Thursday

**Comment by tansaku** [comment]
thanks @rehan-work - I'm away this week so not planning to run the kickoff - but will be back the following week - but hit us up in #localsupport on slack any time :-)

**Comment by tansaku** [comment]
hi @rehan-work - I've pulled this out locally and am changing the `should`s to `is_expected.to`s

**Comment by tansaku** [comment]
I've updated to the develop branch, and I'll merge this as soon as the build goes green if that's okay ...

**Comment by tansaku** [comment]
hmm, CI not running - I'm just going to merge ...

**Comment by tansaku** [comment]
thanks for you work on this - very much appreciated :-)

**Pull request merged**